akd/src/storage/memory.rs

```diff
@@ -155,15 +155,19 @@ impl Storage for AsyncInMemoryDatabase {
                         return Ok(DbRecord::ValueState(item.clone()));
                     }
                 }
-                return Err(StorageError::GetData("Not found".to_string()));
+                return Err(StorageError::NotFound(format!("ValueState {:?}", id)));
             }
         }
         // fallback to regular get/set db
         let guard = self.db.read().await;
         if let Some(result) = (*guard).get(&bin_id).cloned() {
             Ok(result)
         } else {
-            Err(StorageError::GetData("Not found".to_string()))
+            Err(StorageError::NotFound(format!(
```

Comment:
The origin of this error would be really helpful here when debugging. Can we clarify this is erroring out in `get_direct`?

Reply:
We could add log statements, but I specifically excluded "not found" results from log statements as it's a pretty common, expected flow. All of the storage errors which have real failures (i.e not "not found") include a log statement for the error.